pkg/clustermanager/cluster_manager.go

```diff
@@ -399,46 +399,61 @@ func (c *ClusterManager) RunPostCreateWorkloadCluster(ctx context.Context, manag
 }
 
 func (c *ClusterManager) DeleteCluster(ctx context.Context, managementCluster, clusterToDelete *types.Cluster, provider providers.Provider, clusterSpec *cluster.Spec) error {
-	return c.Retrier.Retry(
-		func() error {
-			if clusterSpec.Cluster.IsManaged() {
-				if err := c.PauseEKSAControllerReconcile(ctx, clusterToDelete, clusterSpec, provider); err != nil {
-					return err
-				}
-
-				if clusterSpec.GitOpsConfig != nil {
-					if err := c.DeleteGitOpsConfig(ctx, managementCluster, clusterSpec.GitOpsConfig.Name, clusterSpec.GitOpsConfig.Namespace); err != nil {
-						return err
-					}
-				}
-				if clusterSpec.OIDCConfig != nil {
-					if err := c.DeleteOIDCConfig(ctx, managementCluster, clusterSpec.OIDCConfig.Name, clusterSpec.OIDCConfig.Namespace); err != nil {
-						return err
-					}
-				}
-
-				if clusterSpec.AWSIamConfig != nil {
-					if err := c.DeleteAWSIamConfig(ctx, managementCluster, clusterSpec.AWSIamConfig.Name, clusterSpec.AWSIamConfig.Namespace); err != nil {
-						return err
-					}
-				}
-
-				if err := provider.DeleteResources(ctx, clusterSpec); err != nil {
-					return err
-				}
-
-				if err := c.DeleteEKSACluster(ctx, managementCluster, clusterSpec.Cluster.Name, clusterSpec.Cluster.Namespace); err != nil {
-					return err
-				}
-			}
+	if clusterSpec.Cluster.IsManaged() {
+		if err := c.deleteEKSAObjects(ctx, managementCluster, clusterToDelete, provider, clusterSpec); err != nil {
+			return err
+		}
+	}
 
-			if err := c.clusterClient.DeleteCluster(ctx, managementCluster, clusterToDelete); err != nil {
-				return err
-			}
+	logger.V(1).Info("Deleting CAPI cluster", "name", clusterToDelete.Name)
+	if err := c.clusterClient.DeleteCluster(ctx, managementCluster, clusterToDelete); err != nil {
+		return err
+	}
 
-			return provider.PostClusterDeleteValidate(ctx, managementCluster)
-		},
-	)
+	return provider.PostClusterDeleteValidate(ctx, managementCluster)
+}
+
+func (c *ClusterManager) deleteEKSAObjects(ctx context.Context, managementCluster, clusterToDelete *types.Cluster, provider providers.Provider, clusterSpec *cluster.Spec) error {
```

Comment:
What does the controller workload cluster deletion logic look like? I would imagine running `kubectl delete eksa-cluster` would trigger the chain of deletion from eks-a objects (including gitops, iam, etc.) all the way to the capi objects. If that is the case shouldn't we let controller clean up the resources instead of from CLI?

Reply:
We should let the controller do the deletion yeah, but that's a bigger change. I'm not convinced the controller should delete the other objects but that's a different topic.

In fact we should use the controller for create, upgrade and delete workload clusters and have the logic in one single place. But that's out scope for this first implementation, hopefully a fast follower.